server/controllers/collection.controller/addProjectToCollection.js

```diff
@@ -69,9 +62,19 @@ export default function addProjectToCollection(req, res) {
     ]);
   }
 
-  return Promise.all([collectionPromise, projectPromise])
-    .then(updateCollection)
-    .then(populateReferences)
-    .then(sendSuccess)
-    .catch(sendFailure);
+  try {
+    const collection = await Collection.findById(collectionId).populate(
+      'items.project',
+      '_id'
+    );
+    const project = await Project.findById(projectId);
+
+    const updatedCollection = await updateCollection([collection, project]);
+    if (updatedCollection) {
```

Comment:
What if there’s not an updated collection?  Do we need an `else` here?  We probably also want 404 errors if either the collection or the project are not found.  But I need to take a closer look at this and see what those other functions are doing.

Reply:
we can just send failure message and then return 

```
  if (!updatedCollection) {
    sendFailure(404, 'Collection not updated');
    return;
  }
```

Reply:
for collection and project we can also do the same

    ```
if (!collection) {
      sendFailure(404, 'Collection not found');
      return;
    }
```
```
    if (!project) {
      sendFailure(404, 'Project not found');
      return;
    }
```

Reply:
Ah ok now I'm looking at this properly and I see that the things I was concerned about are already handled in other places.

The not found errors are in https://github.com/processing/p5.js-web-editor/blob/6f5eb26a3afd29615c15be47f5a04c20ae81bed0/server/controllers/collection.controller/addProjectToCollection.js#L17-L25

If `updateCollection` returns `null` then it will have also called `sendFailure` https://github.com/processing/p5.js-web-editor/blob/6f5eb26a3afd29615c15be47f5a04c20ae81bed0/server/controllers/collection.controller/addProjectToCollection.js#L44-L48